From 2ead820f65e2a385be0003d91a56ac61614a1069 Mon Sep 17 00:00:00 2001
From: Jithun Nair <37884920+jithunnair-amd@users.noreply.github.com>
Date: Sat, 19 Oct 2024 02:03:06 -0500
Subject: [PATCH 16/40] HIPCC compilations should also respect
 GLIBCXX_USE_CXX11_ABI flag (#1640)

### Objective

Fixes issues with building PyTorch with `GLIBCXX_USE_CXX11_ABI=0` on
RHEL8, or any other distro where default value of
`GLIBCXX_USE_CXX11_ABI` is 1. Without this fix, `CXX_FLAGS` would
respect `GLIBCXX_USE_CXX11_ABI=0`, but `HIP_CLANG_FLAGS` wouldn't.
---
 cmake/Dependencies.cmake | 1 +
 1 file changed, 1 insertion(+)

diff --git a/cmake/Dependencies.cmake b/cmake/Dependencies.cmake
index 012cad79758..f572f3d07b4 100644
--- a/cmake/Dependencies.cmake
+++ b/cmake/Dependencies.cmake
@@ -1052,6 +1052,7 @@ if(USE_ROCM)
     list(APPEND HIP_CXX_FLAGS -DTHRUST_DEVICE_SYSTEM=THRUST_DEVICE_SYSTEM_HIP)
     list(APPEND HIP_CXX_FLAGS -std=c++17)
     list(APPEND HIP_CXX_FLAGS -DHIPBLAS_V2)
+    list(APPEND HIP_CXX_FLAGS -D_GLIBCXX_USE_CXX11_ABI=${GLIBCXX_USE_CXX11_ABI})
     if(HIP_NEW_TYPE_ENUMS)
       list(APPEND HIP_CXX_FLAGS -DHIP_NEW_TYPE_ENUMS)
     endif()
-- 
2.43.0

